### PR TITLE
docs(testing): SSH-only agent-os native-capture fallback + auth'd-backend harness notes

### PR DESCRIPTION
**Pull request by jmagar**
Techniques discovered building + live-testing the Axon Palette Tauri exe on agent-os when the `agent-os_windows-mcp` upstream wasn't a connected tool.

**desktop-app-testing** — new `references/ssh-fallback-capture.md`: launch + screenshot a native GUI over plain `ssh agent-os` via a `schtasks /it` task in the interactive console session (SSH session 0 has no window station → GUI crashes with `os error 1459`, screenshots blank). Covers `MSYS_NO_PATHCONV=1`, Win32 FindWindow/SetForegroundWindow + SendKeys, forward-slash scp, config seeding, and an in-process-vite + `playwright-core`→Edge dev-loop (token-injecting proxy) for iterating a Tauri/web frontend without a native rebuild. Plus a SKILL.md `## Fallback` pointer + CHANGELOG.

**web-app-testing** — prerequisite note for self-served frontends calling a token-protected backend: bearer-injecting proxy + relative API paths; points at the recipe.

**vibin/agent-os** — `## Connection` gotcha: same session-0-vs-1 / `os error 1459` note + the `schtasks /it` fix.

Live-validated 2026-06-06.

---
## Summary by cubic
Adds SSH-only fallback docs to launch and screenshot native Windows GUIs on `agent-os` when `agent-os_windows-mcp` isn’t reachable, plus guidance for running a web/Tauri frontend against a token-protected backend. Also repairs CI/test failures and enforces dispatch↔MCP boundaries; refreshes generated docs (e.g., `stash.provider.link` is destructive; CLI help defaults to `LAB_PUBLIC_URL`/`LAB_MCP_GATEWAY_URL`).

- **New Features**
  - SSH fallback for native capture: launch via `schtasks /it` in the interactive session; explains session 0 `os error 1459` and PowerShell screenshot flow.
  - Faster loop: run `vite` in-process and drive system Edge via `playwright-core` with a bearer-injecting proxy; frontend must use relative API paths.
  - Cross-links: desktop SKILL fallback pointer; web-app-testing prerequisite for auth’d backends; agent-os SKILL notes and link to the recipe.

- **Bug Fixes**
  - Test harness: add `MockConnectInfo<SocketAddr>` in `lab-auth` oneshot tests to satisfy rate-limit extractors; update stale assertions (marketplace empty `installPath` → not installed; result-format token estimate).
  - Architecture boundaries: move `lab_home` to `dispatch::helpers` with a client re-export; introduce `crate::tool_names` and switch dispatch imports away from `crate::mcp`; allowlist the few expected MCP refs in `architecture_boundaries.rs`.
  - CI/build cleanup: remove dead `acp::sqlite_persistence`, relocate the test ctor into `AcpSessionRegistry<SqliteAcpPersistence>`, make `with_in_process_connector` `pub(crate)`, drop unsupported `expiry-date` keys in `deny.toml`, minor lint fixes; regenerate `docs/generated/*` (destructive flag, proxy env defaults, `logs search` positional query).

<sup>Written for commit 9e221a14fd60510948585a17ccbfea2240528846. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/jmagar/lab/pull/95?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **Documentation**
  * Added SSH-only fallback for capturing native Windows GUIs when normal remote access is unavailable, with session-1 scheduled-task guidance and common gotchas (path/shell quirks).
  * Added guidance for running dev servers against token-protected backends using a bearer-token–injecting proxy and relative API paths.
  * Added a full reference explaining config seeding for homelab backends and a faster frontend dev loop (in-browser proxy + automated Edge-driven iteration) for quicker validation.

### CODE REVIEW

File: plugins/testing/skills/desktop-app-testing/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,18 @@
 # Changelog — desktop-app-testing
 
+## 2026-06-06 — SSH-only fallback
```
[the remark this answers is not in the document]
Comment (jmagar) [reply plugins/testing/skills/desktop-app-testing/CHANGELOG.md:3]: Fixed in 83ed26e0 — added the blank line after the `## 2026-06-06` heading (MD022).

File: plugins/testing/skills/desktop-app-testing/references/ssh-fallback-capture.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
+# SSH fallback: launch + capture a native GUI on agent-os without Windows-MCP
+
+Use when the `agent-os_windows-mcp` upstream is **not reachable from the session** — e.g. the Lab
+gateway is in code-mode and its `execute` interface isn't exposed, or windows-mcp simply isn't a
+connected MCP server. Everything below drives agent-os over plain `ssh agent-os` and was
+live-validated 2026-06 building + capturing the Axon Palette Tauri exe.
+
+## Why plain SSH can't see a GUI (and the fix)
```
[the remark this answers is not in the document]
Comment (jmagar) [reply plugins/testing/skills/desktop-app-testing/references/ssh-fallback-capture.md:8]: Fixed in 83ed26e0 — rewrote the reference with blank lines around every heading and fenced code block (MD022/MD031 clean).

File: plugins/testing/skills/desktop-app-testing/references/ssh-fallback-capture.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  terminal covers it, minimize that terminal — Windows Terminal's class is
+  `CASCADIA_HOSTING_WINDOW_CLASS` → `ShowWindow($hwnd, 6)`.
+- **Type** with `(New-Object -ComObject WScript.Shell).SendKeys(...)` (SendKeys-escape `+ ^ % ~ ( ) { } [ ]`);
+  `^l`/`{ENTER}` etc. work. A frameless window still has a title for `FindWindow` even with
+  `decorations:false`.
+- **Capture** the real desktop: `[System.Windows.Forms.Screen]::PrimaryScreen.Bounds` +
+  `[System.Drawing.Graphics]::FromImage(...).CopyFromScreen(...)` → `bmp.Save("C:\Users\Docker\shot.png")`.
+- Pull it back: `scp 'agent-os:C:/Users/Docker/shot.png' ./` (forward slashes).
+
+Clean up afterwards: `schtasks /delete /tn <name> /f` and stop the process.
```
[the remark this answers is not in the document]
Comment (jmagar) [reply plugins/testing/skills/desktop-app-testing/references/ssh-fallback-capture.md]: Fixed in 83ed26e0 — added `MSYS_NO_PATHCONV=1` to the cleanup `schtasks /delete` command so it matches the launch/run calls.

File: plugins/testing/skills/desktop-app-testing/SKILL.md [hunk 4]
```diff
@@ -27,6 +27,14 @@ lab commit `e87940c0`. If drive calls return `confirmation_required: "...destruc
 gateway predates the fix; rebuild + redeploy (see `references/windows-mcp-calls.md`). Read-only
 `Screenshot`/`Snapshot` are never gated, so a UI/UX *review* works even if the gate isn't lifted.
 
+## Fallback — Windows-MCP not reachable
```
[the remark this answers is not in the document]
Comment (jmagar) [reply plugins/testing/skills/desktop-app-testing/SKILL.md:30]: Fixed in 83ed26e0 — added the blank line after the `## Fallback` heading (MD022).

File: plugins/testing/skills/web-app-testing/CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,11 @@
 # Changelog — web-app-testing
 
+## 2026-06-06 — auth'd-backend prerequisite
```
[the remark this answers is not in the document]
Comment (jmagar) [reply plugins/testing/skills/web-app-testing/CHANGELOG.md:3]: Fixed in 83ed26e0 — added the blank line after the `## 2026-06-06` heading (MD022).

File: plugins/vibin/skills/agent-os/SKILL.md [hunk 6]
```diff
@@ -40,6 +40,8 @@ Windows-MCP is exposed inside agent-os over HTTP + Bearer token and registered i
 
 If the server is unreachable: SSH into `tootie`, confirm the VM container is up (`docker ps --format '{{.Names}}' | grep agent-os`; the container name is `agent-os-win11`), and that the MCP service inside is listening. The container persists `/storage` across restarts so installed software survives.
 
+**SSH-only fallback for a native GUI.** If neither Windows-MCP nor noVNC is reachable but `ssh agent-os` works, you can still launch + screenshot a native GUI — but only from the **interactive console session (session 1)**. Plain SSH lands in **session 0**, which has no window station: a GUI `.exe` launched there crashes with `This operation requires an interactive window station (os error 1459)` and PowerShell `CopyFromScreen` returns a blank frame. Launch + capture via a `schtasks /it` scheduled task instead (prefix `MSYS_NO_PATHCONV=1` so git-bash doesn't mangle the `/flags`); confirm the interactive session with `ssh agent-os 'query user'` (expect `docker console 1 Active`). Full recipe: `~/workspace/lab/plugins/testing/skills/desktop-app-testing/references/ssh-fallback-capture.md`.
```
[the remark this answers is not in the document]
Comment (jmagar) [reply plugins/vibin/skills/agent-os/SKILL.md]: Fixed in 83ed26e0 — changed the absolute `~/workspace/lab/...` path to a repo-relative `plugins/testing/skills/desktop-app-testing/references/ssh-fallback-capture.md` reference.
